File: xml-performance/test-drive-xml-jaxb/src/main/java/com/steelzack/test/drive/jaxb/books/xml/Book.java
package com.steelzack.test.drive.jaxb.books.xml;

import jakarta.xml.bind.annotation.XmlAttribute;
import jakarta.xml.bind.annotation.XmlElement;
import jakarta.xml.bind.annotation.XmlRootElement;

@XmlRootElement(name = "book",
        namespace = "http://www.bookshelexample.com")
public class Book {
    @XmlAttribute(name = "name")
    private String title;
    @XmlElement
    private Integer pages;
    @XmlElement
    private String genre;
    @XmlElement
    private String author;

    public String getAuthor() {
        return author;
    }

    public Integer getPages() {
        return pages;
    }

    public String getGenre() {
        return genre;
    }

    public String getTitle() {
        return title;
    }
}

File: xml-performance/test-drive-xml-jaxb/src/main/kotlin/org/jesperancinha/test/drive/jaxb/books/xml/Book.kt
package org.jesperancinha.test.drive.jaxb.books.xml

import jakarta.xml.bind.annotation.*

@XmlAccessorType(XmlAccessType.FIELD)
@XmlRootElement(name = "book", namespace = "http://www.joaofilipesabinoesperancinha.nl")
open class Book {
    @XmlAttribute(name = "name")
    var title: String? = null

    @XmlElement(name = "pages", namespace = "http://www.joaofilipesabinoesperancinha.nl")
    val pages: Int = 0

    @XmlElement(name = "genre", namespace = "http://www.joaofilipesabinoesperancinha.nl")
    var genre: String? = null

    @XmlElement(name = "author", namespace = "http://www.joaofilipesabinoesperancinha.nl")
    var author: String? = null
}